db_test.go

```diff
@@ -34,6 +33,71 @@ var (
 	err error
 )
 
+func assertErr(t *testing.T, err error, expectErr error) {
+	if expectErr != nil {
+		require.Equal(t, expectErr, err)
+	} else {
+		require.NoError(t, err)
+	}
+}
+
+func removeDir(dir string) {
+	if err := os.RemoveAll(dir); err != nil {
+		panic(err)
+	}
+}
+
+func runNutsDBTest(t *testing.T, opts *Options, test func(t *testing.T, db *DB)) {
+	if opts == nil {
+		opts = new(Options)
```

Comment:
Just confusing, Finally, the opts value is DefaultOptions, why new an Options Object here?

Reply:
Because DefaultOptions is not a pointer, so I new an Options Object, Maybe I should use `opts = &DefaultOptions` to replace it.